src/cs3panel.tsx

```diff
@@ -89,18 +93,67 @@ export class Cs3HeaderWidget extends ReactWidget {
     }
 }
 
+export const Bottom = (props: BottomProps): JSX.Element => {
+
+    const [text, setText] = useState(props.message);
+
+    const onPathOrFileChanged = async (path: string) => {
+        const showHidden: boolean = await props.db.fetch('showHidden') as boolean;
+        if (showHidden == undefined || !showHidden) {
+            const result: Contents.IModel = await requestAPI('/api/contents/' + path, {method: 'get'});
+            const hiddenFilesNo = result.content.filter((file: { name: string; }) => file.name.startsWith('.')).length
+            setText('Hidden files ' + hiddenFilesNo + ' (show)')
+        } else {
+            setText('Cover hidden files')
+        }
+    }
+
+    props.browser.model.pathChanged.connect(async (browser, args) => {
+        onPathOrFileChanged(args.newValue.replace('cs3drive:', ''))
+    })
+
+    props.browser.model.refreshed.connect((browser) => {
+        onPathOrFileChanged(browser.path.replace('cs3drive:', ''))
+    })
+
+    return (
+        <div className={'jp-bottom-div'} onClick={async () => {
+            if (text.startsWith('Hidden files') || text.startsWith('Show hidden')) {
```

Comment:
Don't understand the repetition of this set text nor what we're checking. Shouldn't instead have a boolean "props.showHidden" (or smth) i.e ==true when showing, which sets the corresponding text in a single place?